Products/Zuul/routers/__init__.py

```diff
@@ -174,7 +174,14 @@ def asyncGetTree(self, id=None, additionalKeys=()):
             childData = Marshaller(child).marshal(keys)
             # set children so that there is not an expanding
             # icon next to this child
-            if len(child.children) == 0:
+            # see if there are any subtypes so we can show or not show the
+            # expanding icon without serializing all the children.
+            # Note that this is a performance optimization see ZEN-15857
+            organizer = child._get_object()
+            # this looks at the children's type without waking them up
+            hasChildren = organizer.meta_type in [o['meta_type'] for o in organizer._objects]
```

Comment:
Actually, you don't need a whole list comprehension here - stopping on the first one that matches would be enough. How about an `any` call?